### PR TITLE
Allowing Type option in settings

### PR DESCRIPTION
**Pull request by birdyboy18**
Hey,

I wanted the ability to be able to have content served from the root but use a different layout.

Setting a type option seemed like the best option so I quickly wrote it in. Hopefully it's useful for you too. It's especially useful for repeatable types that you want to hardcode the type in.

I've also updated the readme to reflect this change.

Cheers!

### CODE REVIEW

**Comment by joshmossas** [comment]
Hey @birdyboy18 a type option in the settings is a really great idea. Thanks for the contribution.

Can you explain to me what your thought process was for adding this resolveEntry option? Maybe you could give a use case for where you think it would be useful. If we add this option it'll also need to be included in the documentation.

image

**Comment by birdyboy18** [comment]
Hey @joshmossas ,

The resolve Entry stuff was something I was trying out and I wasn't fully aware that adding it to the top of master would affect this merge request. I will remove and branch off.

I do plan to hopefully have this merged in and documented once I'm done on it. So I'll explain here quick and then let me know if you'd like it logged as an issue, or further explanation.

Sometimes I think you might want to be able to have an entry to entry relationship be boiled down to one specific field. In my case I want to be able to have contentful also manage data for any of my taxonomies. You can't easily relate entries as a taxonmy entry as the generated front-matter isn't compatiable (id, type) with how Hugo works (single value).

Resolving it down to a specific field would mean you can use hugos taxonomy relationships. In my case resolving it to a slug for that taxonomy.

The benefit of doing this is that you can use a content type (category for example) to then create taxonomies and any extra metadata all managed within contentful, and then link any other pieces of content you have as an entry and limit it to just that type.

At the moment the only usecase I can think of would be for allowing Hugo's taxonomy engine to work. 

I hope I've explained that well enough. If not I'll try again. I honstly still feel that Hugo's taxonomies can be really confusing so perhaps I've overengineered something.

I look forward to hearing your inital thoughts and I'm happy to move over into a seperate pull request. Where I can provide better docs and explanation for it.

**Comment by joshmossas** [comment]
@birdyboy18

Oh I gotcha that's a really good idea. Hugo's taxonomy feature can be kind of convoluted at times. Something we we'll need to keep in mind is that in order to retrieve metadata for a taxonomy we need to create a matching markdown file that corresponds with the taxonomy slug. ([hugo docs](https://gohugo.io/content-management/taxonomies/#add-custom-metadata-to-a-taxonomy-term))

So guaranteeing that both the slug listed in the frontmatter and the file path to the generated taxonomy page match up will be an important aspect of getting it right. Perhaps I can create an isTaxonomy options similar to the isHeadless option which creates a folder with all the metadata for that content type. Then the resolveEntry could be used to create a slug in the front matter that references that taxonomy. Or maybe we should go for a more opinionated approach that generates all of the necessary stuff automatically.

**Comment by joshmossas** [comment]
Once your remove the resolveEntry stuff I can go ahead and merge the pull request. If you could also pull from master to get the latest changes as well that would be great.

File: index.js [hunk 1]
```diff
@@ -50,6 +50,8 @@ function initialize() {
                         titleField: types[i].title,
                         dateField: types[i].dateField,
                         mainContent: types[i].mainContent,
+                        type: types[i].type,
+                        resolveEntry: types[i].resolveEntry
```
Comment (joshmossas) [inline index.js]: Remove resolve entry from pull request
```suggestion
```

File: src/getEntryFields.js [hunk 2]
```diff
@@ -1,5 +1,8 @@
-module.exports = entry => {
+module.exports = (entry, resolve = null) => {
```
Comment (joshmossas) [inline src/getEntryFields.js]: remove resolve entry stuff from pull request
```suggestion
module.exports = entry => {
```

File: src/getEntryFields.js [hunk 3]
```diff
@@ -1,5 +1,8 @@
...
     let obj = {};
+    if (resolve) {
```
Comment (joshmossas) [inline src/getEntryFields.js]:
```suggestion
```

File: src/getEntryFields.js [hunk 4]
```diff
@@ -1,5 +1,8 @@
...
+        return entry.fields[resolve.key];
```
Comment (joshmossas) [inline src/getEntryFields.js]:
```suggestion
```

File: src/getEntryFields.js [hunk 5]
```diff
@@ -1,5 +1,8 @@
...
+    }
```
Comment (joshmossas) [inline src/getEntryFields.js]:
```suggestion
```

File: index.js [hunk 6]
```diff
@@ -223,7 +230,8 @@ function getContentType(limit, skip, contentSettings, itemsPulled) {
                                                         break;
                                                     case 'Entry':
                                                         arrayObject = getEntryFields(
-                                                            arrayNode
+                                                            arrayNode,
+                                                            shouldResolveEntry ? contentSettings.resolveEntry : null
```
Comment (joshmossas) [inline index.js]:
```suggestion
```

File: index.js [hunk 7]
```diff
@@ -223,7 +230,8 @@ function getContentType(limit, skip, contentSettings, itemsPulled) {
                                                         break;
                                                     case 'Entry':
                                                         arrayObject = getEntryFields(
-                                                            arrayNode
+                                                            arrayNode,
```
Comment (joshmossas) [inline index.js]:
```suggestion
                                                            arrayNode
```

**Comment by birdyboy18** [comment]
I've now gone back a commit in history and moved that onto a different branch.

Once it's all good I'll set up a new pull request for taxonomy stuff, in the meantime this should handle what it was orignally intended. I've also included the latest changes from master as you asked.

I'm sorry this has taken so long!

Many Thanks

**Comment by joshmossas** [comment]
Thanks @birdyboy18. I no worries I understand. We all get busy.

**Pull request merged**